File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/dependencies/subscope/RepeatUsageDependency_ViewModelTests.cpp
#include <gtest/gtest.h>
#include <memory>
#include "RepeatUsageDependency_ViewModelTestEnvironment.h"
#include "MessageBus.h"
#include <string>
#include "dependencies/Logger.h"
#include "RepeatUsageDependency_ViewModel.h"
#include "RepeatUsageDependency_ViewModelTestEnvironmentImpl.h"

namespace dependencies::subscope
{
  class RepeatUsageDependency_ViewModelTests : public testing::Test
  {
  public:
    std::shared_ptr<RepeatUsageDependency_ViewModel> sut;
    std::shared_ptr<RepeatUsageDependency_ViewModelTestEnvironment> testEnvironment;
    virtual void BuildSut();
    // When Helper Definitions
    virtual void when_OnMessage();
    virtual void when_Log();
  protected:
    void SetUp() override;
  };
  void RepeatUsageDependency_ViewModelTests::SetUp()
  {
    this->testEnvironment = std::make_shared<RepeatUsageDependency_ViewModelTestEnvironmentImpl>();
    this->testEnvironment->Init();
  }
  /// Scenario: Invoke Direct Dependency
  ///   given:
  ///    when: OnMessage
  ///    then:
  TEST_F(RepeatUsageDependency_ViewModelTests, Invoke_Direct_Dependency_given_when_OnMessage_then_)
  {
    this->BuildSut();
    this->when_OnMessage();
  }
  /// Scenario: Invoke Registry Dependency
  ///   given:
  ///    when: Log
  ///    then:
  TEST_F(RepeatUsageDependency_ViewModelTests, Invoke_Registry_Dependency_given_when_Log_then_)
  {
    this->BuildSut();
    this->when_Log();
  }
  void RepeatUsageDependency_ViewModelTests::BuildSut()
  {
    this->sut = this->testEnvironment->BuildSut();
  }
  void RepeatUsageDependency_ViewModelTests::when_OnMessage()
  {
    this->testEnvironment->getMessageBus()->OnMessage(0, { });
  }
  void RepeatUsageDependency_ViewModelTests::when_Log()
  {
    this->testEnvironment->getLogger()->Log(std::string("my message"));
  }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/dependencies/subscope/RepeatUsageDependency_ViewModelTests.cs
/// <filename>
///     RepeatUsageDependency_ViewModelTests.cs
/// </filename>
using dependencies.subscope;
using dependencies;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Collections.Generic;

[Microsoft.VisualStudio.TestTools.UnitTesting.TestClass]
public class RepeatUsageDependency_ViewModelTests
{
    private dependencies.subscope.RepeatUsageDependency_ViewModel sut;

    private dependencies.subscope.RepeatUsageDependency_ViewModelTestEnvironment testEnvironment;

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestInitialize]
    public void SetUp()
    {
        this.testEnvironment = new dependencies.subscope.RepeatUsageDependency_ViewModelTestEnvironmentImpl();
        this.testEnvironment.Init();
    }

    /*

    Scenario: Invoke Direct Dependency
      given:
       when: OnMessage
       then:
   */
    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void Invoke_Direct_Dependency_given_when_OnMessage_then_()
    {
        this.BuildSut();
        this.when_OnMessage();
    }

    /*

    Scenario: Invoke Registry Dependency
      given:
       when: Log
       then:
   */
    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void Invoke_Registry_Dependency_given_when_Log_then_()
    {
        this.BuildSut();
        this.when_Log();
    }

    protected virtual void BuildSut()
    {
        this.sut = this.testEnvironment.BuildSut();
    }

    // When Helper Definitions
    public virtual void when_OnMessage()
    {
        this.testEnvironment.getMessageBus().OnMessage(0, new System.Collections.Generic.List<string> { });
    }

    public virtual void when_Log()
    {
        this.testEnvironment.getLogger().Log("my message");
    }
}
